### ISSUE TITLE
Topic Operator doesn't work properly when ConfigMap name conflicts with a topic name

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#472]
There seems to be a bug in TO caused by the conflict between topic name and topic ConfigMap name. To reproduce, try this scenario:

1) Create following topic:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: my-topic
  labels:
    strimzi.io/kind: topic
    strimzi.io/cluster: my-cluster
data:
  name: npj-topic
  partitions: "3"
  replicas: "1"
  config: |-
    {
      "retention.bytes": "104857600",
      "retention.ms": "86400000",
      "segment.bytes": "10485760"
    }
```
2) Connect into Kafka and create a topic named `my-topic` directly in Kafka
3) Check TO logs for exceptions
4) Give it some time (wait for reconciliation???) or restart TO
5) Check the topics in Kafka: `npj-topic` has been deleted and only `my-topic` remains
6) Check the ConfigMaps: the original config map describing `npj-topic` is still there.

**Comment by TBBle** [comment]
Supposedly fixed by #1331

**Comment by tombentley** [comment]
Thanks @TBBle, this must have slipped through the net.
